core/include/webview/detail/backends/win32_edge.hh

```diff
@@ -681,10 +681,8 @@ protected:
     auto res =
         m_webview->AddScriptToExecuteOnDocumentCreated(wjs.c_str(), &handler);
     if (SUCCEEDED(res)) {
-      // Sadly we need to pump the even loop in order to get the script ID.
-      while (!done) {
-        deplete_run_loop_event_queue();
-      }
+      // Sadly we need to pump the event loop in order to get the script ID.
+      run_event_loop_while([&] { return !done; });
     }
     // TODO: There's a non-zero chance that we didn't get the script ID.
     //       We need to convey the error somehow.
```

Comment:
If the user script was **not** successfully added, we should consider failing fast. Maybe something like:
```c++
else {
  throw exception(WEBVIEW_ERROR_UNSPECIFIED, "Failed to bind script.");
}
```

Reply:
Yes, or we can change the signature of `add_user_script_impl()` to return `result<user_script>` so that we can `return error_info{WEBVIEW_ERROR_UNSPECIFIED, "Failed to add user script"}.

It's a good candidate for a separate PR.